### PR TITLE
Errant GTID Counts metric in VTOrc

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

As pointed out in https://github.com/vitessio/vitess/issues/16828, it would be a good idea to also publish the number of errant GTIDs in each tablet as a metric. This PR accomplishes this goal.

The new metric looks like - 
```
"ErrantGTIDCounts": {"zone1-0000000100": 1},
```

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/16828

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by systay — APPROVED** [review]

File: changelog/21.0/21.0.0/summary.md [hunk 1]
```diff
@@ -137,3 +138,6 @@ Currently many of the configuration options for VReplication Workflows are vttab
 requires restarts of vttablets. We now allow these to be overridden while creating a workflow or dynamically once
 the workflow is in progress. See https://github.com/vitessio/vitess/pull/16583 for details.
 
+### <a id="errant-gtid-metric"/>Errant GTIDs Count Metric
+A new metric called `ErrantGTIDCounts` has been added to the `VTOrc` component.
```
Comment (mattlord) [inline changelog/21.0/21.0.0/summary.md:142]: A count[er] stat is an ever increasing metric that indicates the number of times X has occurred. So I would assume this is the number of times it's had an errant GTID. I would suggest calling this ~ `CurrentErrantGTIDCount` (which is a gauge vs a counter).

File: go/vt/vtorc/inst/instance_dao.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ var forgetAliases *cache.Cache
 var (
 	readTopologyInstanceCounter = stats.NewCounter("InstanceReadTopology", "Number of times an instance was read from the topology")
 	readInstanceCounter         = stats.NewCounter("InstanceRead", "Number of times an instance was read")
+	errantGTIDCounts            = stats.NewGaugesWithSingleLabel("ErrantGTIDCounts", "Number of errant GTIDs in a vttablet", "TabletAlias")
```
Comment (mattlord) [inline go/vt/vtorc/inst/instance_dao.go:64]:
```suggestion
	errantGTIDCounts            = stats.NewGaugesWithSingleLabel("ErrantGTIDCounts", "Number of errant GTIDs a vttablet currently has", "TabletAlias")
```

**Review by mattlord — APPROVED** [review]

**Review by timvaillancourt — APPROVED** [review]

**Review by deepthi — APPROVED** [review]
LGTM, but it will be nice to change the summary as suggested by Matt.

File: changelog/21.0/21.0.0/summary.md (hunk 1 above)
Comment (GuptaManan100) [reply changelog/21.0/21.0.0/summary.md]: Fixed

File: go/vt/vtorc/inst/instance_dao.go (hunk 2 above)
Comment (GuptaManan100) [reply go/vt/vtorc/inst/instance_dao.go]: Fixed

**Pull request merged**